blocks/bricks/recurrent/base.py

```diff
@@ -46,6 +46,9 @@ def initial_states(self, batch_size, *args, **kwargs):
             The keyword arguments of the application call.
 
         """
+        if not hasattr(self, 'apply') or not self.apply.states:
+            return
+
```

Comment:
@rizar I think this check could be carried out in `Brick.__init__` method. I reverted back the changes in `BaseRecurrent`.